freeipa/src/main/java/com/sequenceiq/freeipa/flow/freeipa/upscale/action/FreeIpaUpscaleActions.java

```diff
@@ -244,7 +244,8 @@ private void validateResourceResults(StackContext context, Exception exception,
                 }
                 List<CloudResourceStatus> templates = results.stream().filter(result -> CommonResourceType.TEMPLATE == result.getCloudResource().getType()
                         .getCommonResourceType()).collect(Collectors.toList());
-                if (!templates.isEmpty() && (templates.get(0).isFailed() || templates.get(0).isDeleted())) {
+                if (!templates.isEmpty()
+                        && (templates.stream().anyMatch(CloudResourceStatus::isFailed) || templates.stream().anyMatch(CloudResourceStatus::isDeleted))) {
```

Comment:
simplify a bit:
```
templates.stream().anyMatch(template -> template.isDeleted() || template.isFailed())
```
checking emptiness is not necessary in case of streams, and you don't have to initiate 2 stream to evaluate